src/Payments.sol

```diff
@@ -479,28 +479,49 @@ contract Payments is
         Account storage payee = accounts[rail.token][rail.to];
 
         uint256 oldRate = rail.paymentRate;
+        bool isTerminated = isRailTerminated(rail);
 
         // Settle the payer's lockup to account for elapsed time
         uint256 lockupSettledUpto = settleAccountLockup(payer);
 
         // Validate rate changes based on rail state and account lockup
-        validateRateChangeRequirements(
-            rail,
-            payer,
-            oldRate,
-            newRate,
-            lockupSettledUpto,
-            oneTimePayment
-        );
+        if (isTerminated) {
+            if (block.number >= maxSettlementEpochForTerminatedRail(rail)) {
+                return
+                    modifyPaymentForTerminatedRailBeyondLastEpoch(
+                        rail,
+                        newRate,
+                        oneTimePayment
+                    );
+            }
+
+            require(
+                newRate <= oldRate,
+                "failed to modify rail: cannot increase rate on terminated rail"
+            );
+        } else {
+            validateRateChangeRequirementsForNonTerminatedRail(
+                rail,
+                payer,
+                oldRate,
+                newRate,
+                lockupSettledUpto
+            );
+        }
 
         // --- Settlement Prior to Rate Change ---
         handleRateChangeSettlement(railId, rail, oldRate, newRate);
 
         // Calculate the effective lockup period
-        uint256 effectiveLockupPeriod = calculateEffectiveLockupPeriod(
-            rail,
-            payer
-        );
+        uint256 effectiveLockupPeriod;
+        if (isTerminated) {
+            effectiveLockupPeriod = remainingEpochsForTerminatedRail(rail);
+        } else {
+            effectiveLockupPeriod = isRailInDebt(rail, payer)
```

Comment:
This rail cannot be in-debt (enforced by `validateRateChangeRequirementsForNonTerminatedRail`).